### ISSUE TITLE
[Support]: Frigate (mostly) does not start - 'timed out'.

### ISSUE THREAD

**Issue by jpwheless** [blakeblackshear/frigate#4610]
### Describe the problem you are having

Hello,

I've had Frigate successfully up and running on a Linux and Windows system previously with the same config, but I just moved it to its permanent Ubuntu Server home and I'm having issues getting it started consistently.  Once it does start it runs without issues.

The log file below shows it failing to start once and then succeeding on the retry.  On the line where it says 'timed out', it sometimes but not always mentions MQTT.  The MQTT server is on a different system, but I can ping and port scan with no issues from this host and connect an MQTT client from a desktop on the same subnet.  The MQTT server is set to allow anonymous logins, so user/pw shouldn't matter and hasn't with other clients.

The default ufw firewall on this host is disabled for now and the install is all default other than Docker.

Any troubleshooting ideas?

I appreciate the support.  Frigate is pretty awesome, too.

### Version

0.11.1

### Frigate config file

```yaml
# DOCKER COMPOSE FILE
#########################
version: "3"
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: blakeblackshear/frigate:0.11.1
    shm_size: "64mb" # update for your cameras based on calculation above
    volumes:
      - ./filesystem/config/config.yml:/config/config.yml:ro
      - ./filesystem/media:/media/frigate
    ports: # External port:container port
      - "5001:5000" # 5000 was in use by something else.
      - "1935:1935" # RTMP feeds
    environment:
      - FRIGATE_RTSP_PASSWORD= "password"

# FRIGATE CONFIG FILE
# Note: removed all but one camera, object settings, etc for min length
#########################
mqtt:
  host: mrdisky.localdomain
  port: 1883
  user: user
  password: 'password'

detectors:
  cpu1:
    type: cpu
    num_threads: 4

record:
  enabled: True
  events:
    pre_capture: 5 # seconds before the event to include
    post_capture: 5 # seconds after the event to include (default: shown below)
    retain:
      default: 10
      mode: active_objects
      objects:
        person: 15

ffmpeg:
  output_args:
    record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a copy


snapshots:
  enabled: True
  clean_copy: True
  timestamp: False
  bounding_box: True
  crop: True
  required_zones: [zone_front_walkway, zone_driveway, zone_street]
  retain:
    default: 10
    objects:
      person: 15

cameras:
  ipcam_porch: # Camera name
    ffmpeg:
      inputs:
        - path: rtsp://user:password@ipcam_porch.localdomain:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
            - rtmp
        - path: rtsp://user:password@ipcam_porch.localdomain:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
    rtmp:
      enabled: true # <-- RTMP should be disabled if your stream is not H264
    detect:
      width: 704
      height: 480
      fps: 5 # Try to reduce to minimum required for the purpose.  5 fps is default.
      enabled: True
    objects:
      filters:
        person:
          mask:
            - 192,268,251,269,254,226,194,222 # Flamingo decorations
    zones:
      zone_front_walkway:
        coordinates: 704,480,704,0,226,198,198,258,0,256,0,480
```

### Relevant log output

```shell
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2022-12-05 05:14:40] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate    | Starting migrations
frigate    | [2022-12-05 05:14:40] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2022-12-05 05:14:40] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | timed out
frigate    | [cmd] python3 exited 1
frigate    | [cont-finish.d] executing container finish scripts...
frigate    | [cont-finish.d] done.
frigate    | [s6-finish] waiting for services.
frigate    | [s6-finish] sending all processes the TERM signal.
frigate    | [s6-finish] sending all processes the KILL signal and exiting.
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2022-12-05 05:15:41] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate    | Starting migrations
frigate    | [2022-12-05 05:15:41] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2022-12-05 05:15:41] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | [2022-12-05 05:15:42] detector.cpu1                  INFO    : Starting detection process: 216
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Output process started: 218
frigate    | [2022-12-05 05:15:42] ws4py                          INFO    : Using epoll
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Camera processor started for ipcam_porch: 220
frigate    | [2022-12-05 05:15:42] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Camera processor started for ipcam_driveway: 224
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Camera processor started for ipcam_gable: 225
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Capture process started for ipcam_porch: 226
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Capture process started for ipcam_driveway: 228
frigate    | [2022-12-05 05:15:42] frigate.app                    INFO    : Capture process started for ipcam_gable: 229
frigate    | [2022-12-05 05:15:42] ws4py                          INFO    : Using epoll
```

### FFprobe output from your camera

```shell
From the host:
  built with gcc 12 (Ubuntu 12.2.0-1ubuntu1)
  configuration: --prefix=/usr --extra-version=1ubuntu1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-shared
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Dahua

**Comment by jpwheless** [comment]
I swapped all of my hostnames to IPs and it looks like I'm getting the same error:

```
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2022-12-06 14:52:26] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate    | [2022-12-06 14:52:26] frigate.app                    INFO    : Creating directory: /tmp/cache
frigate    | Starting migrations
frigate    | [2022-12-06 14:52:26] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2022-12-06 14:52:26] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | timed out
frigate    | [cmd] python3 exited 1
frigate    | [cont-finish.d] executing container finish scripts...
frigate    | [cont-finish.d] done.
frigate    | [s6-finish] waiting for services.
frigate    | [s6-finish] sending all processes the TERM signal.
frigate    | [s6-finish] sending all processes the KILL signal and exiting.
```

But if I enter the wrong address for the MQTT server, I still get that error:

```
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2022-12-06 14:58:05] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate    | [2022-12-06 14:58:05] frigate.app                    INFO    : Creating directory: /tmp/cache
frigate    | Starting migrations
frigate    | [2022-12-06 14:58:05] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2022-12-06 14:58:05] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | timed out
frigate    | [cmd] python3 exited 1
frigate    | [cont-finish.d] executing container finish scripts...
frigate    | [cont-finish.d] done.
frigate    | [s6-finish] waiting for services.
frigate    | [s6-finish] sending all processes the TERM signal.
frigate    | [s6-finish] sending all processes the KILL signal and exiting.
```

I'm still able to connect to the MQTT server with MQTT Explorer on a desktop, and ping the server from the Frigate host, while I'm testing different Frigate configs.

**Comment by jpwheless** [comment]
Same basic log, but this one shows that lone 'timed out' error occurring before the MQTT client timing out.  Is there something besides MQTT trying to start up at that moment that might be having issues, or are the two timeout messages both from the MQTT client?

```
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2022-12-08 16:03:36] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate    | [2022-12-08 16:03:36] frigate.app                    INFO    : Creating directory: /tmp/cache
frigate    | Starting migrations
frigate    | [2022-12-08 16:03:36] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2022-12-08 16:03:36] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | timed out
frigate    | [2022-12-08 16:03:41] frigate.mqtt                   ERROR   : Unable to connect to MQTT server: timed out
frigate    | [cmd] python3 exited 1
frigate    | [cont-finish.d] executing container finish scripts...
frigate    | [cont-finish.d] done.
frigate    | [s6-finish] waiting for services.
frigate    | [s6-finish] sending all processes the TERM signal.
frigate    | [s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by NickM-27** [comment]
There are lots of things that start but the only things that will cause frigate to fail & stop are edgetpu not being available (there is a long error message, so this isn't that), the database being locked, and mqtt not working.

It is odd that it shows that error, I have not seen that and don't know where that comes from. If you are using the correct MQTT ip then do you see frigate publishing any topics when watching with mqtt explorer?

**Comment by jpwheless** [comment]
Good idea.  I watched with MQTT Explorer while trying to start Frigate and I didn't see any new messages being published, so either Frigate isn't able to connect or it terminates before any messages are able to be published.

I tried a few more things:
- Disallowed anonymous login on the MQTT server and gave Frigate the user/password.  No change in behavior.
- Removed the MQTT user/password from the Frigate config.  No error message saying it couldn't connect.
- Installed MQTTX CLI on the Frigate host machine.  It was able to publish a message to the MQTT server without any issues.
- Started an Ubuntu Docker container on the Frigate host machine and installed MQTTX CLI on it.  It was also able to publish a message to the MQTT server without issues.
- Reduced the Frigate config file to the minimum:
```
mqtt:
  host: mrdisky.localdomain
cameras:
  back:
    ffmpeg:
      inputs:
        - path: rtsp://user:password@ipcam_driveway.localdomain:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 704
      height: 480
```
No changes in behavior with the minimal config.

I feel like I've ruled out MQTT as the source of the issue at this point, but I could be wrong.

**Comment by NickM-27** [comment]
Your example doesn't have user/pass implemented for mqtt, is that correct? and I was able to reproduce this `timed out` log on 0.11 with a broken mqtt link so I do believe it is the issue.

**Comment by jpwheless** [comment]
Thank you for trying the broken MQTT link.  Any ideas how the Frigate MQTT client could be failing while all the other tests I've done have successfully connected?

The example config was from one of the attempts without a user/pw.  Here's with (and with IPs instead of a hostnames):
```
mqtt:
  host: 192.168.50.60
  user: user
  password: 'password'
cameras:
  back:
    ffmpeg:
      inputs:
        - path: rtsp://user:password@192.168.51.10:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 704
      height: 480
```
I just did some wireshark captures on the host machine with a couple different filters:
`tshark -f "dst 192.168.50.60 or 192.168.51.10 and not arp"`
`tshark -f "port 1883"`
The 192.168.51.10 is the camera IP, to see if Frigate tries to talk to that also.  1883 is the MQTT server port, of course.

First with an MQTTX test to make sure wireshark was working properly, which produced these packets:
```    2 2.879321675 192.168.50.246 → 192.168.50.60 TCP 74 54182 → 1883 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=4111869449 TSecr=0 WS=128
    3 2.879536830 192.168.50.246 → 192.168.50.60 TCP 66 54182 → 1883 [ACK] Seq=1 Ack=1 Win=64256 Len=0 TSval=4111869449 TSecr=930351111
    4 2.885457257 192.168.50.246 → 192.168.50.60 MQTT 97 Connect Command
    5 2.885835091 192.168.50.246 → 192.168.50.60 TCP 66 54182 → 1883 [ACK] Seq=32 Ack=12 Win=64256 Len=0 TSval=4111869455 TSecr=930351117
    6 2.888441664 192.168.50.246 → 192.168.50.60 MQTT 89 Publish Message [test_mqttx], Disconnect Req
    7 2.889340613 192.168.50.246 → 192.168.50.60 TCP 66 54182 → 1883 [FIN, ACK] Seq=55 Ack=13 Win=64256 Len=0 TSval=4111869459 TSecr=930351120
```
192.168.50.246 is the Frigate host's IP.  Neither capture command showed MQTT or camera traffic while trying to start the Frigate container.

**Comment by NickM-27** [comment]
Is there some firewall or security rule keeping it from communicating? I can't see any reason why that'd be the case. 

Another thing to try would be create a local MQTT container in this docker compose and see if it'll run with that.

**Comment by jpwheless** [comment]
Pretty sure I figured out the cause while trying to set up mosquitto on the Frigate host machine like you said.  Ubuntu Server offers to install Docker via snap when you first install it, and it looks like snap either causes issues with Docker or has extra security permissions that were preventing Frigate from contacting the mqtt server.  I was having trouble getting anything to connect to the local mosquitto container also.

I removed the snap docker and reinstalled it with apt and it's started up reliably each time I've tried it.

Thanks for the troubleshooting help!
